### PR TITLE
RFC: Some degree trigonometric functions, `sind`, `cosd`, `tand`, `asind` , `acosd`, `asecd`, `acsd`, `acotd`, `atand` accept a square matrix.

### LINKED ISSUE #651: implement degree trig functions on square matrices
See this closed PR: https://github.com/JuliaLang/julia/pull/27254.

### LINKED ISSUE #32876: implement degree trig functions on square matrices
See this closed PR: https://github.com/JuliaLang/julia/pull/27254.

### PR DESCRIPTION
**Pull request by AtsushiSakai**
I improved some degree trigonometric functions, `sind`, `cosd`, `tand`, `asind`, `acosd`, `asecd`, `acscd`, `acotd` to accept an square matrix to fix JuliaLang/LinearAlgebra.jl#651.
Also, I added some tests, updated docs of these functions, updated NEWS.md, and generalized a unit convert function as a for loop element which is proposed in https://github.com/JuliaLang/julia/pull/27254#issuecomment-391896477.

### CODE REVIEW

**Comment by AtsushiSakai** [comment]
I close/reopen to rerun CI.

**Comment by AtsushiSakai** [comment]
I think the CI error is not related this PR.

```
Error During Test at /buildworker/worker/tester_linux32/build/share/julia/test/cmdlineargs.jl:211
  Test threw exception
  Expression: read(`$exename -e $code`, String) == string(cpu_threads)
  failed process: Process(`/buildworker/worker/tester_linux32/build/bin/julia -Cnative -J/buildworker/worker/tester_linux32/build/lib/julia/sys.so --depwarn=error --check-bounds=yes -g1 --startup-file=no --startup-file=no --color=no -e 'print(Threads.nthreads())'`, ProcessSignaled(11)) [0]
 
```

**Comment by ViralBShah** [comment]
@stefankarpinski @simonbyrne Could one of you (or someone else) take a look?

**Comment by AtsushiSakai** [comment]
I fixed a conflict of News.md. PTAL.

**Comment by ViralBShah** [comment]
I do want another set of eyes on this. Trying @dkarrasch @StefanKarpinski @simonbyrne.

**Comment by ViralBShah** [comment]
Why do we not do this for all the trig functions? For example - `atand`?

File: base/special/trig.jl [hunk 1]
```diff
@@ -1294,5 +1302,5 @@ end
 
 Compute the inverse tangent of `y` or `y/x`, respectively, where the output is in degrees.
 """
-atand(y)    = rad2deg(atan(y))
-atand(y, x) = rad2deg(atan(y,x))
+atand(y)    = rad2deg.(atan(y))
```
Comment (AtsushiSakai) [inline base/special/trig.jl:1305]: @ViralBShah Thank you for your comment. The PR title and news.md was missing `atand`, but it supports a square matrix by this change.

File: test/math.jl [hunk 2]
```diff
@@ -341,6 +341,35 @@ end
     @test Array(acosh.(STAA)) == acosh.(TAA)
     @test Array(acsch.(STAA)) == acsch.(TAA)
     @test Array(acoth.(STAA)) == acoth.(TAA)
+    @test sind(TAA) == sin(deg2rad.(TAA))
+    @test cosd(TAA) == cos(deg2rad.(TAA))
+    @test tand(TAA) == tan(deg2rad.(TAA))
+    @test asind(TAA) == rad2deg.(asin(TAA))
+    @test acosd(TAA) == rad2deg.(acos(TAA))
+    @test atand(TAA) == rad2deg.(atan(TAA))
```
Comment (AtsushiSakai) [inline test/math.jl:349]: @ViralBShah `atand` is tested here.

**Pull request merged**